### ISSUE TITLE
Addic7ed tarpitting?

### ISSUE THREAD

**Issue by burkasaurusrex** [morpheus65535/bazarr#861]
**Describe the bug**
It looks like Addict7ed may be tarpitting overusage by IP address or something? I've got anti-captcha enabled with a valid account. Addic7ed will work for a few days when I access it from a new IP address, then there will be these constant log messages:

```
17/03/2020 00:26:51|WARNING |retry.api                       |HTTPSConnectionPool(host='www.addic7ed.com', port=443): Read timed out. (read timeout=10), retrying in 5 seconds...|
17/03/2020 00:27:06|WARNING |retry.api                       |HTTPSConnectionPool(host='www.addic7ed.com', port=443): Read timed out. (read timeout=10), retrying in 5 seconds...|
17/03/2020 00:27:34|WARNING |retry.api                       |HTTPSConnectionPool(host='www.addic7ed.com', port=443): Read timed out. (read timeout=10), retrying in 5 seconds...|
17/03/2020 00:28:25|WARNING |retry.api                       |HTTPSConnectionPool(host='www.addic7ed.com', port=443): Read timed out. (read timeout=10), retrying in 5 seconds...|
17/03/2020 00:29:27|WARNING |retry.api                       |HTTPSConnectionPool(host='www.addic7ed.com', port=443): Read timed out. (read timeout=10), retrying in 5 seconds...|
17/03/2020 00:29:43|WARNING |retry.api                       |HTTPSConnectionPool(host='www.addic7ed.com', port=443): Read timed out. (read timeout=10), retrying in 5 seconds...|
17/03/2020 00:30:22|WARNING |retry.api                       |HTTPSConnectionPool(host='www.addic7ed.com', port=443): Read timed out. (read timeout=10), retrying in 5 seconds...|
17/03/2020 00:31:04|WARNING |retry.api                       |HTTPSConnectionPool(host='www.addic7ed.com', port=443): Read timed out. (read timeout=10), retrying in 5 seconds...|
17/03/2020 00:32:07|WARNING |retry.api                       |HTTPSConnectionPool(host='www.addic7ed.com', port=443): Read timed out. (read timeout=10), retrying in 5 seconds...|
17/03/2020 00:35:49|WARNING |retry.api                       |HTTPSConnectionPool(host='www.addic7ed.com', port=443): Read timed out. (read timeout=10), retrying in 5 seconds...|
17/03/2020 00:36:35|WARNING |retry.api                       |HTTPSConnectionPool(host='www.addic7ed.com', port=443): Read timed out. (read timeout=10), retrying in 5 seconds...|
17/03/2020 00:36:51|WARNING |retry.api                       |HTTPSConnectionPool(host='www.addic7ed.com', port=443): Read timed out. (read timeout=10), retrying in 5 seconds...|
17/03/2020 00:49:53|WARNING |retry.api                       |HTTPSConnectionPool(host='www.addic7ed.com', port=443): Read timed out. (read timeout=10), retrying in 5 seconds...|
17/03/2020 00:52:45|WARNING |retry.api                       |HTTPSConnectionPool(host='www.addic7ed.com', port=443): Read timed out. (read timeout=10), retrying in 5 seconds...|
17/03/2020 00:54:27|WARNING |retry.api                       |HTTPSConnectionPool(host='www.addic7ed.com', port=443): Read timed out. (read timeout=10), retrying in 5 seconds...|
17/03/2020 00:54:43|WARNING |retry.api                       |HTTPSConnectionPool(host='www.addic7ed.com', port=443): Read timed out. (read timeout=10), retrying in 5 seconds...|
17/03/2020 00:59:09|WARNING |retry.api                       |HTTPSConnectionPool(host='www.addic7ed.com', port=443): Read timed out. (read timeout=10), retrying in 5 seconds...|
17/03/2020 01:00:53|WARNING |retry.api                       |HTTPSConnectionPool(host='www.addic7ed.com', port=443): Read timed out. (read timeout=10), retrying in 5 seconds...|
17/03/2020 01:02:34|WARNING |retry.api                       |HTTPSConnectionPool(host='www.addic7ed.com', port=443): Read timed out. (read timeout=10), retrying in 5 seconds...|
17/03/2020 01:04:56|WARNING |retry.api                       |HTTPSConnectionPool(host='www.addic7ed.com', port=443): Read timed out. (read timeout=10), retrying in 5 seconds...|
17/03/2020 01:06:36|WARNING |retry.api                       |HTTPSConnectionPool(host='www.addic7ed.com', port=443): Read timed out. (read timeout=10), retrying in 5 seconds...|
17/03/2020 01:07:10|WARNING |retry.api                       |HTTPSConnectionPool(host='www.addic7ed.com', port=443): Read timed out. (read timeout=10), retrying in 5 seconds...|
17/03/2020 01:13:22|WARNING |retry.api                       |HTTPSConnectionPool(host='www.addic7ed.com', port=443): Read timed out. (read timeout=10), retrying in 5 seconds...|
17/03/2020 01:19:35|WARNING |retry.api                       |HTTPSConnectionPool(host='www.addic7ed.com', port=443): Read timed out. (read timeout=10), retrying in 5 seconds...|
17/03/2020 01:21:50|WARNING |retry.api                       |HTTPSConnectionPool(host='www.addic7ed.com', port=443): Read timed out. (read timeout=10), retrying in 5 seconds...|
17/03/2020 01:25:47|WARNING |retry.api                       |HTTPSConnectionPool(host='www.addic7ed.com', port=443): Read timed out. (read timeout=10), retrying in 5 seconds...|
17/03/2020 01:33:39|WARNING |retry.api                       |HTTPSConnectionPool(host='www.addic7ed.com', port=443): Read timed out. (read timeout=10), retrying in 5 seconds...|
17/03/2020 01:34:15|WARNING |retry.api                       |HTTPSConnectionPool(host='www.addic7ed.com', port=443): Read timed out. (read timeout=10), retrying in 5 seconds...|
17/03/2020 01:34:55|WARNING |retry.api                       |HTTPSConnectionPool(host='www.addic7ed.com', port=443): Read timed out. (read timeout=10), retrying in 5 seconds...|
17/03/2020 01:35:39|WARNING |retry.api                       |HTTPSConnectionPool(host='www.addic7ed.com', port=443): Read timed out. (read timeout=10), retrying in 5 seconds...|
17/03/2020 01:37:02|WARNING |retry.api                       |HTTPSConnectionPool(host='www.addic7ed.com', port=443): Read timed out. (read timeout=10), retrying in 5 seconds...|
17/03/2020 01:37:32|WARNING |retry.api                       |HTTPSConnectionPool(host='www.addic7ed.com', port=443): Read timed out. (read timeout=10), retrying in 5 seconds...|
17/03/2020 01:39:11|WARNING |retry.api                       |HTTPSConnectionPool(host='www.addic7ed.com', port=443): Read timed out. (read timeout=10), retrying in 5 seconds...|
17/03/2020 01:40:15|WARNING |retry.api                       |HTTPSConnectionPool(host='www.addic7ed.com', port=443): Read timed out. (read timeout=10), retrying in 5 seconds...|
17/03/2020 01:47:40|WARNING |retry.api                       |HTTPSConnectionPool(host='www.addic7ed.com', port=443): Read timed out. (read timeout=10), retrying in 5 seconds...|
17/03/2020 01:48:44|WARNING |retry.api                       |HTTPSConnectionPool(host='www.addic7ed.com', port=443): Read timed out. (read timeout=10), retrying in 5 seconds...|
17/03/2020 01:49:22|WARNING |retry.api                       |HTTPSConnectionPool(host='www.addic7ed.com', port=443): Read timed out. (read timeout=10), retrying in 5 seconds...|
```
If I swap IP addresses it will work again. Also, if I access Addic7ed from a web browser on another device on the network, I can't access the site. I can easily log into the site from my phone on a cellular connection, etc.

**Software (please complete the following information):**
 - Bazarr: 0.8.4.2
 - OS: Unraid (Docker)

**Additional context**
I have Addic7ed enabled with valid creds, "Random User-Agents" turned on, and an active Anti-Captcha account key.

**Comment by morpheus65535** [comment]
Agree that it looks like tarpitting. What do you expect Bazarr to do? Circumvent this and continue hammering the site? We must, to an extend, respect those providers limitation. When they'll unblock your IP, you'll be able to download again from them.

**Comment by burkasaurusrex** [comment]
Yea, good question. Maybe treat time-outs similar to other connection errors for other providers and start to space out connection attempts to try and avoid getting blocked?

**Comment by morpheus65535** [comment]
@halali you are more familiar than me with provider throttling. Could you take a look please?

**Comment by halali** [comment]
I will try to catch HTTPSConnectionPool exception and raise exception that is allowed for throttling... I just dont remember how it is called right now

**Comment by burkasaurusrex** [comment]
Captured some more logs in case it's helpful: https://pastebin.com/3Dz3hhJt

It does look like a retry is happening every 5-10 seconds right now.

**Comment by morpheus65535** [comment]
@halali did you had time to look into this one?

**Comment by Jqh63** [comment]
Same message error about a timeout and retry using Addic7ed.
Had to disable my Bazarr container.

**Comment by redglory** [comment]
Some providers block your IP for hammering or double authentication (same login with different IPs to avoid share).

@morpheus65535 you could implement some kind of exception (i.e LockedIPAddress) which could be configured based on each provider. For instance, LegendasDivx unlocks your IP at midnight (if I remember correctly). Then, we could define custom times on "bazarr/get_providers.py".

**Comment by morpheus65535** [comment]
@redglory Yes I've seen that in your latest PR. If you're willing to give it a try, I would gladly accept another PR to implement this exception. :-)

**Comment by redglory** [comment]
@morpheus65535 I've implemented a new Exception "IPAddressBlocked" on libs/subliminal_patch/exceptions.py and reference that on bazarr/get_providers.py 

I'm just testing this before push to dev branch

**Comment by morpheus65535** [comment]
@burkasaurusrex could you try dev branch? @redglory just pushed a fix that will throttle addic7ed for one hour instead of hammering the provider.

**Comment by ExorDG** [comment]
I am getting the same warning message:
`12/05/2020 20:19:55|WARNING |retry.api                       |HTTPSConnectionPool(host='www.addic7ed.com', port=443): Max retries exceeded with url: /login.php (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x05275898>, 'Connection to www.addic7ed.com timed out. (connect timeout=10)')), retrying in 5 seconds...|`

I guess I will wait for them to unblock my IP address. Good to know that I am not the only one.

**Comment by redglory** [comment]
> I am getting the same warning message:
> `12/05/2020 20:19:55|WARNING |retry.api |HTTPSConnectionPool(host='www.addic7ed.com', port=443): Max retries exceeded with url: /login.php (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x05275898>, 'Connection to www.addic7ed.com timed out. (connect timeout=10)')), retrying in 5 seconds...|`
> 
> I guess I will wait for them to unblock my IP address. Good to know that I am not the only one.

Are you running the master or develop branch ?

**Comment by morpheus65535** [comment]
Should be fixed by @redglory latest fix in dev branch. Reopen if this one happen again.
